test/testother.cpp

```diff
@@ -6053,6 +6055,27 @@ class TestOther : public TestFixture {
               "}", "test.c");
         ASSERT_EQUALS("", errout.str());
     }
+
+    void testUnsignedLessThanZero() {
+        check("struct d {\n"
+              "  unsigned n;\n"
+              "};\n"
+              "void f(void) {\n"
+              "  struct d d;\n"
+              "  d.n = 3;\n"
+              "\n"
+              "  if (d.n < 0) {\n"
+              "    return;\n"
+              "  }\n"
+              "\n"
+              "  if (0 > d.n) {\n"
+              "    return;\n"
+              "  }\n"
+              "}", "test.c");
+        ASSERT_EQUALS("[test.c:8]: (style) Checking if unsigned variable 'd.n' is less than zero.\n"
```

Comment:
Oh, I was looking at another test file.
Looks like using `tok->astOperand1()->expressionString()` fixed this, there are no parentheses in warning.